sh/drive-utils.sh

```diff
@@ -0,0 +1,535 @@
+        printf "%s\n" "Name: $(printf "%s\n" "${json_collect_file_info}" | _json_value name 1 1)"
+        printf "%s\n" "ID: ${FILE_ID}"
+        printf "%s\n\n" "Type: $(printf "%s\n" "${json_collect_file_info}" | _json_value mimeType 1 1)"
+    )"
+    printf "%s\n" "${info_collect_file_info}" >> "${LOG_FILE_ID}"
+}
+
+###################################################
+# Error logging wrapper
+###################################################
+_error_logging() {
+    "${QUIET:-_print_center}" "justify" "Upload ERROR" ", ${1:-} not ${STRING:-uploaded}." "=" 1>&2
+    printf "\n\n\n" 1>&2
+}
+
+###################################################
+# Get information for a gdrive folder/file.
+# Globals: 2 variables, 1 function
+#   Variables - API_URL, API_VERSION
+#   Functions - _json_value
+# Arguments: 3
+#   ${1} = folder/file gdrive id
+#   ${2} = information to fetch, e.g name, id
+#   ${3} = Access Token
+# Result: On
+#   Success - print fetched value
+#   Error   - print "message" field from the json
+# Reference:
+#   https://developers.google.com/drive/api/v3/search-files
+###################################################
+_drive_info() {
+    [ $# -lt 3 ] && printf "Missing arguments\n" && return 1
+    folder_id_drive_info="${1}" fetch_drive_info="${2}" token_drive_info="${3}"
+    unset search_response_drive_info
+
+    "${EXTRA_LOG:-:}" "justify" "Fetching info.." "-" 1>&2
+    search_response_drive_info="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
+        -H "Authorization: Bearer ${token_drive_info}" \
+        "${API_URL}/drive/${API_VERSION}/files/${folder_id_drive_info}?fields=${fetch_drive_info}&supportsAllDrives=true" || :)" && "${CURL_PROGRESS_EXTRA_CLEAR:-:}" 1 1>&2
+    "${EXTRA_LOG_CLEAR:-:}" 1 1>&2
+
+    printf "%s\n" "${search_response_drive_info}"
+    return 0
+}
+
+###################################################
+# Search for an existing file on gdrive with write permission.
+# Globals: 2 variables, 2 functions
+#   Variables - API_URL, API_VERSION
+#   Functions - _url_encode, _json_value
+# Arguments: 3
+#   ${1} = file name
+#   ${2} = root dir id of file
+#   ${3} = Access Token
+# Result: print file id else blank
+# Reference:
+#   https://developers.google.com/drive/api/v3/search-files
+###################################################
+_check_existing_file() (
+    [ $# -lt 3 ] && printf "Missing arguments\n" && return 1
+    name_check_existing_file="${1##*/}" rootdir_check_existing_file="${2}" token_check_existing_file="${3}"
+    unset query_check_existing_file search_response_check_existing_file id_check_existing_file
+
+    "${EXTRA_LOG:-:}" "justify" "Checking if" " file exists.." "-" 1>&2
+    query_check_existing_file="$(_url_encode "name='${name_check_existing_file}' and '${rootdir_check_existing_file}' in parents and trashed=false and 'me' in writers")"
+
+    search_response_check_existing_file="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)